src/rollback_to_stable/rts.c

```diff
@@ -116,22 +116,40 @@ __wt_rts_btree_apply_all(WT_SESSION_IMPL *session, wt_timestamp_t rollback_times
     WT_CURSOR *cursor;
     WT_DECL_RET;
     WT_TIMER timer;
-    uint64_t rollback_count, rollback_msg_count;
+    uint64_t max_count, rollback_count, rollback_msg_count;
     char ts_string[WT_TS_INT_STRING_SIZE];
     const char *config, *uri;
+    bool have_cursor;
 
     __wt_timer_start(session, &timer);
-    rollback_count = 0;
+    max_count = rollback_count = 0;
     rollback_msg_count = 0;
 
+    /*
+     * Walk the metadata first to count how many files we have overall. That allows us to give
+     * signal about progress.
+     */
     WT_RET(__wt_metadata_cursor(session, &cursor));
+    have_cursor = true;
     while ((ret = cursor->next(cursor)) == 0) {
-        /* Log a progress message. */
-        __wt_rts_progress_msg(session, &timer, rollback_count, &rollback_msg_count, false);
-        ++rollback_count;
+        WT_ERR(cursor->get_key(cursor, &uri));
+        if (WT_BTREE_PREFIX(uri))
+            ++max_count;
```

Comment:
An earlier investigation of the time to process many files with RTS [here](https://jira.mongodb.org/browse/WT-10863?focusedCommentId=5433886&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-5433886) shows that it can take 60 seconds to process 1 million tables. So performing the metadata search twice shouldn't cause much overhead.